src/coreclr/src/vm/readytoruninfo.h

```diff
@@ -20,15 +20,40 @@ typedef DPTR(struct READYTORUN_SECTION) PTR_READYTORUN_SECTION;
 
 class PrepareCodeConfig;
 
+typedef DPTR(class ReadyToRunCoreInfo) PTR_ReadyToRunCoreInfo;
+class ReadyToRunCoreInfo
+{
+private:
+    PTR_PEImageLayout               m_pLayout;
+    PTR_READYTORUN_CORE_HEADER      m_pCoreHeader;
+    
+public:
+    ReadyToRunCoreInfo();
+    ReadyToRunCoreInfo(PEImageLayout * pLayout, READYTORUN_CORE_HEADER * pCoreHeader);
+
+    PEImageLayout * GetLayout() const { return m_pLayout; }
+    IMAGE_DATA_DIRECTORY * FindSection(ReadyToRunSectionType type) const;
+
+    PTR_PEImageLayout GetImage() const
+    {
+        LIMITED_METHOD_CONTRACT;
+        return m_pLayout;
+    }
+};
+
 typedef DPTR(class ReadyToRunInfo) PTR_ReadyToRunInfo;
 class ReadyToRunInfo
 {
     friend class ReadyToRunJitManager;
 
     PTR_Module                      m_pModule;
-
-    PTR_PEImageLayout               m_pLayout;
     PTR_READYTORUN_HEADER           m_pHeader;
+    bool                            m_isComponentAssembly;
+    DomainCompositeImage*           m_compositeImage;
+    ReadyToRunInfo*                 m_compositeInfo;
+
+    ReadyToRunCoreInfo              m_component;
+    const ReadyToRunCoreInfo*       m_composite;
```

Comment:
A nit - in the runtime, we tend to prefix pointer variable names with `p`, so for example this one would be m_pComposite.

Reply:
Will do - I was originally under the impression that this is generally a remnant of the no longer used Hungarian notation.